DfE.FindInformationAcademiesTrusts/Pages/Trusts/Academies/Details.cshtml.cs

```diff
@@ -1,17 +1,29 @@
 using DfE.FindInformationAcademiesTrusts.Data;
+using Microsoft.AspNetCore.Mvc;
 
 namespace DfE.FindInformationAcademiesTrusts.Pages.Trusts.Academies;
 
 public class AcademiesDetailsModel : TrustsAreaModel, IAcademiesAreaModel
 {
     public IOtherServicesLinkBuilder LinkBuilder { get; }
 
-    public AcademiesDetailsModel(ITrustProvider trustProvider, IOtherServicesLinkBuilder linkBuilder) :
-        base(trustProvider, "Academies in this trust")
+    public AcademiesDetailsModel(ITrustProvider trustProvider, IDataSourceProvider dataSourceProvider,
+        IOtherServicesLinkBuilder linkBuilder) :
+        base(trustProvider, dataSourceProvider, "Academies in this trust")
     {
         PageTitle = "Academies details";
         LinkBuilder = linkBuilder;
     }
+    public override async Task<IActionResult> OnGetAsync()
+    {
+        var pageResult = await base.OnGetAsync();
+
+        if (pageResult.GetType() == typeof(NotFoundResult)) return pageResult;
 
+        var giasSource = await GetGiasDataUpdated();
+        DataSources = new[] { new DataSourceListEntry(giasSource!, "Details")
```

Comment:
We should use `!` very sparingly - if there's no chance that giasSource is ever null then our method `GetGiasDataUpdated` should not return a nullable. Conversely, if there is a chance that null is returned then we should handle it properly as `!` will throw an exception. This also applies to the other Get data sources